test/lib/sdkdriver/src/transport/websocket/client.ts

```diff
@@ -0,0 +1,100 @@
+import Transport from '../';
+import { EventEmitter } from 'events';
+import * as WebSocket from 'ws';
+
+/**
+ * WebSocket client {@link Transport}.
+ * @class
+ * @fires Transport#close
+ * @fires Transport#message
+ */
+export default class WSClientTransport extends EventEmitter implements Transport {
+  private readonly  _deps: any;
+  private readonly _sendBuffer: Array<string>;
+  private readonly _wsUrl: string;
+  private _wsClient: WebSocket | null;
+
+  /**
+   * Constructor.
+   * @param {string} wsUrl - WebSocket URL
+   */
+  constructor(wsUrl: string, deps: any = {}) {
+    super();
+    this._deps = {
+      WebSocket,
+      ...deps
+    };
+    this._sendBuffer = [];
+    this._wsClient = null;
+    this._wsUrl = wsUrl;
+  }
+
+  /**
+   * Close the {@link WSClientTransport}.
+   * @returns {void}
+   */
+  close(): void {
+    if (!this._wsClient) {
+      return;
+    }
+    try {
+      this._wsClient.close();
```

Comment:
I think this is only supposed to throw if `code` or `reason` are provided: https://html.spec.whatwg.org/multipage/web-sockets.html#dom-websocket-close

Do we need the `try`/`catch`?